### PR TITLE
ETH-615 Create Network subgraph entity + ETH-708 Add operator.isEligibleToVote to subgraph

### PR DESCRIPTION
**Pull request by samigabor**

### CODE REVIEW

**Comment by samigabor** [comment]
[Add operator.isEligibleToVote to subgraph](https://linear.app/streamr/issue/ETH-708/add-operatoriseligibletovote-to-subgraph)

File: packages/network-subgraphs/src/operator.ts [hunk 1]
```diff
@@ -11,9 +12,23 @@ import {
     QueuedDataPayout
 } from '../generated/templates/Operator/Operator'
 import { loadOrCreateDelegation, loadOrCreateDelegator, loadOrCreateDelegatorDailyBucket,
+    loadOrCreateNetwork,
     loadOrCreateOperator, loadOrCreateOperatorDailyBucket } from './helpers'
 import { QueueEntry } from '../generated/schema'
 
+/** Delegated is used for tracking the total amount delegated across all Operators */
+export function handleDelegated(event: Delegated): void {
+    let operatorContract = event.address.toHexString()
```
Comment (samt1803) [inline packages/network-subgraphs/src/operator.ts:21]: remove unused values

File: packages/network-subgraphs/src/operator.ts [hunk 2]
```diff
@@ -11,9 +12,23 @@ import {
...
+    let delegator = event.params.delegator.toHexString()
+    let newDelegation = event.params.amountDataWei
+    log.info('handleDelegated: operatorContract={} delegator={} amountDataWei={}', [
+        operatorContract, delegator, newDelegation.toString()])
+
+    let network = loadOrCreateNetwork()
+    network.totalDelegation = network.totalDelegation.plus(newDelegation)
```
Comment (samt1803) [inline packages/network-subgraphs/src/operator.ts:28]: rename to totalDelegated or cumulativeTotalDelegated

**Comment by samt1803** [comment]
add totalDelegated, totalUndelegated cumulative values

File: packages/network-subgraphs/src/operator.ts [hunk 3]
```diff
@@ -124,6 +140,9 @@ export function handleOperatorValueUpdate(event: OperatorValueUpdate): void {
     bucket.valueWithoutEarnings = operator.valueWithoutEarnings
     bucket.totalStakeInSponsorshipsWei = operator.totalStakeInSponsorshipsWei
     bucket.save()
+
+    let network = loadOrCreateNetwork()
+    network.totalStake = network.totalStake.plus(stakeChange)
```
Comment (samt1803) [inline packages/network-subgraphs/src/operator.ts:145]: save() missing

File: packages/network-subgraphs/schema.graphql [hunk 4]
```diff
@@ -224,6 +221,55 @@ type ProjectStakingDayBucket @entity {
...
+  id: ID! # StreamrConfig contract address
+
+  "Total DATA staked - slashed in all sponsorships"
+  totalStake: BigInt!
+  "Total DATA delegated to all operators"
+  totalDelegation: BigInt!
+  "The number of all Sponsorships ever created through the SponsorshipFactory"
+  sponsorshipsCount: Int!
+  "The number of all Sponsorships ever created through the SponsorshipFactory that are currently paying"
+  payingSponsorshipsCount: Int!
```
Comment (jtakalai) [inline packages/network-subgraphs/schema.graphql:234]:
```suggestion
  "The number of all Sponsorships ever created through the SponsorshipFactory that are currently funded"
  fundedSponsorshipsCount: Int!
```

**Review by jtakalai — CHANGES_REQUESTED** [review]

**Review by samt1803 — CHANGES_REQUESTED** [review]

**Review by jtakalai — CHANGES_REQUESTED** [review]
Idea: at https://github.com/streamr-dev/network-contracts/pull/754/files#diff-39b6cb2399417bc5d4d2bca0a173da5e4504765b46f60344dd8d62dd168d73d3R183 then maybe if flag result was no-kick, then copy the protection end timestamp into flag-target's Operator entity

File: packages/network-subgraphs/src/operator.ts (hunk 3 above)
Comment (samigabor) [reply packages/network-subgraphs/src/operator.ts:145]: Added

File: packages/network-subgraphs/src/operator.ts (hunk 1 above)
Comment (samigabor) [reply packages/network-subgraphs/src/operator.ts:21]: Removed

File: packages/network-subgraphs/src/operator.ts (hunk 2 above)
Comment (samigabor) [reply packages/network-subgraphs/src/operator.ts]: renamed

**Comment by samigabor** [comment]
> add totalDelegated, totalUndelegated cumulative values

renamed totalDelegation and added totalUndelegated

**Review by jtakalai — APPROVED** [review]

**Pull request merged**